hw/top_earlgrey/data/top_earlgrey.c.tpl

```diff
@@ -16,3 +16,16 @@ const ${helper.plic_sources().name.as_c_type()}
   [${irq_id_name.as_c_enum()}] = ${source_name.as_c_enum()},
 % endfor
 };
+
+/**
+ * Alert Handler Alert Id to Peripheral Map
+ *
+ * This array is a mapping from `${helper.alert_alerts().name.as_c_type()}` to
+ * `${helper.alert_sources().name.as_c_type()}`.
+ */
+const ${helper.alert_alerts().name.as_c_type()}
+    ${helper.alert_mapping_name.as_snake_case()}[${len(helper.alert_mapping())}] = {
+% for (alert_id_name, source_name) in helper.alert_mapping().items():
```

Comment:
Reading the code below, I think that `alert_mapping` is a `dict`, but then its items don't come out in a defined order, which might be a problem because we check in the generated files.

Reply:
Yeah this is something I fixed up later.